components/Masonry.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <div class="masonry">
+    <div
+      v-for="image in images"
+      :key="image.url"
+      class="masonry__item"
+    >
+      <img
+        :src="image.url"
+        class="masonry__image"
+      >
+      <user-badge
+        :user="image.user"
+        class="masonry__button masonry__button--user"
+      />
+      <like-button
+        :count="image.like.count"
+        class="masonry__button masonry__button--like"
+      />
+      <v-btn
+        class="btn--likecoin masonry__button masonry__button--use"
+        color="secondary"
+      >
+        Use Image
+      </v-btn>
+    </div>
+  </div>
+</template>
+
+<script>
+import LikeButton from '~/components/LikeButton';
+import UserBadge from '~/components/UserBadge';
+
+export default {
+  name: 'masonry',
```

Comment:
Prefer `masonry-images-grid` for fulfilling https://vuejs.org/v2/style-guide/#Multi-word-component-names-essential